lib/utils/matrix_sdk_extensions/event_extension.dart

```diff
@@ -158,4 +165,66 @@ extension LocalizedBody on Event {
       room.isDirectChat ||
       !isSameSenderWith(nextEvent) ||
       type == EventTypes.Encrypted;
+
+  bool get isPinned => room.pinnedEventIds.contains(eventId);
+
+  Future<bool> unpin() async {
```

Comment:
If you unpin fail how do you error handling?

Reply:
IMO: It's a bad idea, why don't write an interactor?

Reply:
`UpdatePinnedMessagesInteractor`